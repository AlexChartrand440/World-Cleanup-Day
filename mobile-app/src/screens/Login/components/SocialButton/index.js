export { default as SocialButton } from './SocialButton';
export { default as styles } from './styles';
