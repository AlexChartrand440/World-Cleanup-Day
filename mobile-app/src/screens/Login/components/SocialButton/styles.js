import { colors } from '../../../../themes';
import {
  getHeightPercentage,
  getWidthPercentage,
} from '../../../../shared/helpers';

export default {
  $iconSize: 20,
  container: {
    backgroundColor: colors.$white,
    flexDirection: 'row',
    borderRadius: 4,
    paddingVertical: 8,
    paddingHorizontal: 5,
    alignItems: 'center',
  },
  iconContainer: {
    flex: -1,
    height: getHeightPercentage(25),
    width: getWidthPercentage(25),
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden',
    marginLeft: 10,
  },
  icon: {
    alignItems: 'center',
    justifyContent: 'center',
    resizeMode: 'contain',
    height: 25,
    width: 25,
    margin: 5,
  },
  text: {
    fontFamily: 'Lato-Bold',
    fontSize: 17,
    fontWeight: 'bold',
    textAlign: 'center',
    flex: 1,
    color: colors.$white,
  },
};

