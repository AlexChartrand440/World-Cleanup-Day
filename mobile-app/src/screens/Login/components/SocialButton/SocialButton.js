import React from 'react';
import { TouchableOpacity, Text, View, Image } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const SocialButton = ({ icon, color, style = {}, text,
  onPress, backgroundColor = 'black' }) => {
  const textStyles = [styles.text];
  const iconContainerStyles = [styles.iconContainer];
  if (color) {
    iconContainerStyles.push({ backgroundColor: color });
  }

  return (
    <TouchableOpacity onPress={onPress} activeOpacity={0.7}>
      <View style={[styles.container, style, { backgroundColor }]}>
        <View style={iconContainerStyles}>
          <Image style={styles.icon} source={icon} />
        </View>
        <Text style={textStyles}>{text}</Text>
      </View>
    </TouchableOpacity>
  );
};

SocialButton.defaultProps = {
  style: {},
};

SocialButton.propTypes = {
  style: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  icon: PropTypes.number.isRequired,
};
export default SocialButton;
