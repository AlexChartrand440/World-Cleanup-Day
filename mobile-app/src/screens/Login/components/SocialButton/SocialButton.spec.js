import React from 'react';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import SocialButton from './SocialButton';

describe('Testing SocialButton', () => {
  it('renders correctly', () => {
    const onPress = () => {};
    const tree = renderer
      .create(<SocialButton icon="facebook" color="white" text="Example" onPress={onPress} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('reacts to press', () => {
    const onPress = sinon.spy();
    const wrapper = shallow(
      <SocialButton icon="facebook" color="white" text="Example" onPress={onPress} />,
    );
    wrapper.simulate('press');

    expect(wrapper.find(TouchableOpacity)).toHaveLength(1);
    expect(onPress.calledOnce).toBe(true);
  });
});
