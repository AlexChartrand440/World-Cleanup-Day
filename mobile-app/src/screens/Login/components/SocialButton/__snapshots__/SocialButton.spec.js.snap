// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SocialButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        undefined,
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "backgroundColor": "white",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          Object {
            "color": "white",
          },
        ]
      }
    >
      Example
    </Text>
  </View>
</View>
`;
